### PR TITLE
Refactoring i18n package to accommodate new UI modules in cleaner and moduler way

### PR DESCRIPTION
**Pull request by ppraja22**
*Adding new modules in i18n package seems cumbersome*

*Made changes in the existing i18n package to have translation files at folder level in-place of having it at parent (dictionaries) level. Ultimately for new UI modules , we will create new folder under parent(dictionaries) folder and put all the related translation files and default text to that folder*

_Current :-_
```
--dictionaries
   |
   -- All Translation file
```

_Post this PR :-_
```
--dictionaries
   |
   --authenticator
      |
      --all translation files and default texts related to authenticator
   --new XYZ UI module
      |
      --all translation files and default texts related to this new module
```

Tested by changing the translation values in existing component as well by adding and changing the values in new UI module's components (which is what is being targeted to accommodate from this PR)"

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/ui/src/i18n/dictionaries/index.ts [hunk 1]
```diff
@@ -1,7 +1,15 @@
...
-export { enDict } from './en';
-export { esDict } from './es';
-export { frDict } from './fr';
-export { itDict } from './it';
-export { jaDict } from './ja';
-export { zhDict } from './zh';
+import * as authenticatorDict from './authenticator';
+
+//merge all the new module translations in respective locale constants
+export const deDict = { ...authenticatorDict.deDict };
```
Comment (ericclemmons) [inline packages/ui/src/i18n/dictionaries/index.ts:4]: Just adding this for my own understanding: this change is enabling this in the future:

```suggestion
export const deDict = {
  ...authenticatorDict.deDict,
  ...someOtherDict.deDict
};
```

File: packages/ui/src/i18n/translations.ts [hunk 2]
```diff
@@ -33,63 +34,7 @@ import {
...
-  SIGNING_IN_BUTTON: 'Signing in',
-  SKIP: 'Skip',
-  SUBMIT: 'Submit',
-  SUBMITTING: 'Submitting',
-  VERIFY_CONTACT: 'Verify Contact',
-  VERIFY_HEADING: 'Account recovery requires verified contact information',
-  VERIFY: 'Verify',
-  WEBSITE: 'Website',
-} as const;
+export const DefaultTexts = { ...defaultTexts } as const;
```
Comment (ericclemmons) [inline packages/ui/src/i18n/translations.ts:37]: @wlee221 Are `DefaultTexts` an API we want exposed for customers?  I noticed that it was already exported, but we don't document it:
> https://ui.docs.amplify.aws/components/authenticator#internationalization-i18n

**Review by ericclemmons — APPROVED** [review]
This looks good, just making sure I didn't misunderstand anything!

File: packages/ui/src/i18n/translations.ts (hunk 2 above)
Comment (wlee221) [reply packages/ui/src/i18n/translations.ts:37]: No, i don't think we found a strong reason to expose it yet. I found it more intuitive to just write the translatable string directly than having to look through these `DefaultTexts` map.

**Review by wlee221 — APPROVED** [review]

**Pull request merged**